lua/entities/gmod_wire_expression2/core/custom/prop.lua

```diff
@@ -454,23 +478,233 @@ end
 
 --------------------------------------------------------------------------------
 
+e2function void bone:boneManipulate(vector pos, angle rot, isFrozen, gravity, collision)
+	local ent, index = boneVerify(self, this)
+	if not ValidAction(self, ent, "manipulate", this, index) then return end
+	
+	setPos(this, pos)
+	setAng(this, rot)
+	
+	this:EnableMotion(isFrozen == 0)
+	this:EnableGravity(gravity ~= 0)
+	this:EnableCollisions(collision ~= 0)
+	
+	ent:PhysWake()
+end
+
+e2function void bone:boneFreeze(isFrozen)
+	if not boneVerify(self, this) then return end
+	this:EnableMotion( isFrozen == 0 )
+	this:Wake()
+end
+
+__e2setcost(30)
+
+e2function void bone:setCollisions(enable)
+	if not boneVerify(self, this) then return end
+	this:EnableCollisions(enable ~= 0)
+	this:Wake()
+end
+
+e2function void bone:setDrag( number drag )
+	if not boneVerify(self, this) then return end
+	this:EnableDrag( drag ~= 0 )
+end
+
+e2function void bone:setInertia( vector inertia )
+	if not boneVerify(self, this) then return end
+	if inertia:IsZero() then return end
+	this:SetInertia(inertia)
+end
+
+e2function void bone:setBuoyancy(number buoyancy)
+	if not boneVerify(self, this) then return end
+	this:SetBuoyancyRatio( math.Clamp(buoyancy, 0, 1) )
+end
+
+e2function void bone:setPhysicalMaterial(string material)
+	if not boneVerify(self, this) then return end
+	if not ValidAction(self, this, "physmat") then return end
+	this:SetMaterial(material)
+end
+
+e2function void bone:setVelocity(vector velocity)
+	local ent, index = boneVerify(self, this)
+	if not ValidAction(self, ent, "velocitynxt", this, index) then return end
+	this:SetVelocity(velocity)
+	ent:PhysWake()
+end
+
+e2function void bone:setVelocityInstant(vector velocity)
+	local ent, index = boneVerify(self, this)
+	if not ValidAction(self, ent, "velocityins", this, index) then return end
+	this:SetVelocityInstantaneous(velocity)
+	ent:PhysWake()
+end
+
+e2function void bone:setAngVelocity(vector velocity)
+	local ent, index = boneVerify(self, this)
+	if not ValidAction(self, ent, "angvelnxt", this, index) then return end
+	this:SetAngleVelocity(velocity)
+	ent:PhysWake()
+end
+
+e2function void bone:setAngVelocityInstant(vector velocity)
+	local ent, index = boneVerify(self, this)
+	if not ValidAction(self, this, "angvelinst", ent, index) then return end
+	this:SetAngleVelocityInstantaneous(velocity)
+	ent:PhysWake()
+end
+
+__e2setcost(500)
+
+-- This code was leveraged from Garry's Mod. Perhaps it would be a bit cleaner with a slight rewrite.
+e2function void entity:makeStatue(enable)
+	if not ValidAction(self, this, "statue") then return end
+	if this:GetNWBool("IsStatue") and enable ~= 0 then return
+	elseif not this:GetNWBool("IsStatue") and enable == 0 then return end
+	
+	
+	local timeout = this.statueTimeout
+	if timeout and timeout.time > CurTime() then
+		self:throw("Wait before repeated statue operations", false)
```

Comment:
`self:throw` returns the second argument you pass it, so you can just do return `self:throw("...", nil)`